### DISCUSSION TITLE
Optimize PNG of the mods

### DISCUSSION THREAD

**Discussion by FrannDzs** [FakeFishGames/Barotrauma#12190] [Ideas and suggestions]
I found that many mods have images at an incorrect size (odd), far exceeding 4096 and/or uncompressed, so this would solve the problem for some people using texture compression. 
Also making sure a texture does not exceed 4096 is important for performance reasons. 

1. **Add an option in the game graphics settings to permanently correct the PNG sizes of installed mods**

- If any texture exceeds 4096 in width or height, reduce their size to that limit keeping the original aspect ratio but being multiple of 4.

- If any texture is equal to or less than 4096 in width or height, rounding their size to the nearest multiple of 4 keeping the original aspect ratio.

So that the game can compress these textures correctly.

2. **Add an option in the graphics settings of the game to permanently compress the png's of the installed mods using the current method or some advanced algorithm (like pngquant) without the loss of quality being visible, (e.g. compressing at a high quality ratio).**

**Comment by Regalis11** [comment]
While this could be a useful change since a lot of modders seem to just completely ignore the console warnings about incorrect texture sizes, automatically changing the sizes of the textures would misalign the sprites in that mod.

**Reply by FrannDzs** [reply]
I know, but one or two pixels makes a difference?...rounding to the nearest should not have much impact... 

I've been doing some tests. 
- If the image does not exceed 4096 on any of its sides, the rounding will be minimal and I believe it has no impact.
- If the images exceed 4096 there will be errors, because the size change is very large. **Maybe this can be addressed in a different way**.

**Reply by Udrakan** [reply]
Yes, 1px does matter for an item sprite in baro. It's noticable and annoying when I misalign stuff by accident.
